### ISSUE TITLE
Freezing one monitor

### ISSUE THREAD

**Issue by Vilkasek** [godotengine/godot#78267]
### Godot version

v4.0.2.stable.arch_linux

### System information

Manjaro, laptop with additional external monitor

### Issue description

I've got problem if I'm using Godot on my Manjaro with my second monitor, my second monitor freezing some times. I'm using nvidia gpu in laptop and intel igpu with hibrid prime drivers. The same issue is when I using dedicated GPU and iGPU, but on the second egnine can crash.

### Steps to reproduce

- Do something.
- Saving
- Writing script
- Imporrting

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
> Manjaro, laptop with additional external monitor

Please provide more details, as this could very likely be driver or hardware related

**Comment by Vilkasek** [comment]
Okay, that issue can be avoided if I'm using only one monitor, but that's not good, cause my laptop have very small screen

**Comment by Vilkasek** [comment]
> > Manjaro, laptop with additional external monitor
> 
> Please provide more details, as this could very likely be driver or hardware related

My laptop is Acer Nitro 5 with 32 GB RAM, 3 disks - 2 NVMe SSDs and 1 SATa HDD, I've got i5-9300H processor and Nvidia geForce GTX 1650 with 4 GB of VRAM.

**Comment by Vilkasek** [comment]
I've got fully upgraded Manjaro on board with GNOME 44 DE

**Comment by Vilkasek** [comment]
Godot is installed on default directory via pacman and project is on secondary NVMe with 1 TB provided by SiliconPower

**Comment by Vilkasek** [comment]
If I turn off my main (external) monitor after freeze I can working on integrated laptop's monitor and If I turn on again main monitor for few minutes I can work normally, but after 2 maybe 4 minutes engine start lagging and after that freeze whole monitor.

Oh, I forgot my Nvidia driver is NVIDIA 530.41.03, OpenGL API 3.3.0, and Vulkan API 1.3.236. Project using Forward+ redering.

**Comment by AThousandShips** [comment]
Please edit your comments instead of posting several comments in a row, anyone subscribed to this repository will be pinged for each comment

**Comment by Vilkasek** [comment]
Okey, sorry for spaming with pings

**Comment by AThousandShips** [comment]
It's okay! But keep it in mind

**Comment by Vilkasek** [comment]
I'm using X11

**Comment by hrstwn** [comment]
Hi. I also experiencing the same thing. I have tried the regular executable from https://godotengine.org/ and the flatpak version from Flathub. Both output the same issue. It runs just when on just my laptop screen. 

Here's my system info


`Godot v4.1.1.stable - elementary OS 7.1 Horus 7.1 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2060 (nvidia; 535.113.01) - AMD Ryzen 7 4800H with Radeon Graphics (16 Threads)`

Not sure if it's related, but my second screen is a Huion Kamvas 22 Plus
and my laptop is Hp Omen 15 running elementary OS 7.1 with X11

here's the system details from the Os's settings:

```
8 × AMD Ryzen 7 4800H with Radeon Graphics 
AMD® Renoir
NVIDIA® GeForce® RTX® 2060 Mobile
16 GB memory
1 TB storage
```

**Comment by mapopi-mm** [comment]
I'm having a similar issue but it doesn't get to crash, just a noticeable slowdown when using a second monitor that only happens when I left click. Recorded two videos showing the difference, both are from the same instance of the game, only difference is that one has the second monitor on and the other doesn't (Unplugged)

https://github.com/godotengine/godot/assets/132522372/03f9613c-c8db-40eb-af07-b8535276e6aa
Video with 2nd monitor on

https://github.com/godotengine/godot/assets/132522372/6f9d2e0b-6721-4a5d-94e3-dbe60322a0a3
Without 2nd monitor

My second monitor is a Huion Kamvas 16 at 60hz, primary is an ASUS VG32VQ1B at 165hz. Running ubuntu 23.04 on X11. System config:

```
Ryzen 5 7600
RTX 2070
16 GB DDR5 Ram
```

**Comment by SylvainBigonneau** [comment]
Yup, having the same issue of my external monitor freezing, on my ubuntu 23.10 . I have to trigger a refresh (either by unplugging, turning of, or changing parameters in display settings) to get it to work again

AMD Ryzen™ 5 5600H with Radeon™ Graphics × 12
NVIDIA GeForce RTX™ 3070 Laptop GPU / AMD Radeon™ Graphics
16,0 GiB RAM

**Comment by SylvainBigonneau** [comment]
Update: I have upgraded my Ubuntu to 23.10 yesterday, and so far the freezing issues seem to be fixed.

**Comment by kornatskyi** [comment]
Have the same issue on 
OS: Linux Mint 21.2 x86_64 
Laptop: Lenovo Legion 5 
Kernel: 5.15.0-88-generic 
GPU: NVIDIA GeForce GTX 1660 Ti Mobile
I'm using 2 identical external monitors, but only one of freezes, always the same one. It happens after I'm launching the main scene from the basic 2D tutorial from GoDot docs. 
Have tried installing different drivers and switching DE from Cinnamon to GNOME, didn't help.

**Comment by Calinou** [comment]
> (Godot or the OS seems to switch GPUs when loading a scene?!):

The project manager and editor are separate processes. The project manager always uses the Compatibility rendering method unless forced on the command line (since it starts faster), while the editor uses the rendering method specified in the project settings.

**Comment by jobesu14** [comment]
I cannot reproduce those freezes / slow down when using compatbility mode.
Everything works perfectly on Godot 4.1 when using Compatibility rendering mode.

**Comment by fsoft72** [comment]
I am having the same issue on my laptop + external monitor.
If I run in Compatibility mode, everything is fine.

My laptop is: MSI Vector GP78HX 13VI-425IT, RTX 4090  
I am running on:  Kubuntu 23.10 64bit with nvidia drivers, my session is X11
Godot Version: 4.2.1 official from website

The freeze happens almost immediately, but only on the external monitor it seems like a FrameBuffer freeze issue.
The laptop monitor continues to work and I am able to logout / login again resetting the external monitor as well

**Comment by Calinou** [comment]
@fsoft72 Can you reproduce this on 4.1.2? If not, please test [4.2](https://downloads.tuxfamily.org/godotengine/4.2/) dev, beta and RC builds as well to determine when the regression started.

**Comment by fsoft72** [comment]
Hello @Calinou 

I downloaded some versions from: https://downloads.tuxfamily.org/godotengine/

Unfortunately, all the version I tested had this error. 
Godot 4.0.2 used to work on my machine, but it was some time ago, so I suspect the nvidia driver or other elements have changed since my last run (I am still learning Godot)

I tested (with the usual problem) these versions:

- Godot_v4.0-stable_linux.x86_64
- Godot_v4.0.2-stable_linux.x86_64
- Godot_v4.0.4-stable_linux.x86_64
- Godot_v4.1-stable_linux.x86_64
- Godot_v4.1.2-stable_linux.x86_64
- Godot_v4.3-dev2_linux.x86_64

If you need me to test a specific version, please let me know

**Comment by PatNowak** [comment]
I have the same. My details:

```
Godot Engine v4.2.1.stable.official.b09f793f5 - https://godotengine.org
Vulkan API 1.3.242 - Forward Mobile - Using Vulkan Device #2: NVIDIA - NVIDIA GeForce GTX 1050 Ti

# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.4 LTS
Release:	22.04
Codename:	jammy
```

**Comment by notJunius** [comment]
Hey i am having this issue with 4.3, has anyone been able to solve this?

**Comment by enekonieto** [comment]
@notJunius there is a workaround solving the issue to some people (including me):

` Project -> Project settings... -> Rendering -> Renderer -> Rendering Method: gl_compatibility`

**Comment by notJunius** [comment]
Yeah it works fine if i do that, but I wish it would work on forward+, is there going to be a ton of issues if i spend my time editing dual monitor in compatibility and then switch to forward+ when exporting?

> [@notJunius](https://github.com/notJunius) there is a workaround solving the issue to some people (including me):
> 
> ` Project -> Project settings... -> Rendering -> Renderer -> Rendering Method: gl_compatibility`

**Comment by notJunius** [comment]
> Yeah it works fine if i do that, but I wish it would work on forward+, is there going to be a ton of issues if i spend my time editing dual monitor in compatibility and then switch to forward+ when exporting?
> 
> > [@notJunius](https://github.com/notJunius) there is a workaround solving the issue to some people (including me):
> > ` Project -> Project settings... -> Rendering -> Renderer -> Rendering Method: gl_compatibility`

also thank you for replying to me it helps a lot, i am glad to at least be able to do dual monitor on compatability now

**Comment by Calinou** [comment]
>  is there going to be a ton of issues if i spend my time editing dual monitor in compatibility and then switch to forward+ when exporting?

Visuals are different between Forward+/Mobile and Compatibility in 3D, so you should make sure to test with Forward+ regularly.

You can use [feature tags](https://docs.godotengine.org/en/latest/tutorials/export/feature_tags.html) to override the rendering method for the editor only, while keeping the project running from the editor (or the exported project) unaffected.

**Comment by nkbs33** [comment]
Hi, I had the same problem. I solved it by setting the Nvidia graphic card mode to performance, which is on demand by default. Hope this works for you guys.
